subsys/net/l2/ethernet/arp.c

```diff
@@ -445,9 +445,31 @@ struct net_pkt *net_arp_prepare(struct net_pkt *pkt,
 	return pkt;
 }
 
-static inline void arp_update(struct net_if *iface,
-			      struct in_addr *src,
-			      struct net_eth_addr *hwaddr)
+static void gratuitous_arp(struct net_if *iface,
```

Comment:
can you rename it to arp_gratuitous ?

Reply:
That sounds very awkward actually as typically this support is called "gratuitous ARP", so naming the function same way is quite practical.

Reply:
We can certainly change the name, I have no issues with that. It is just I do not fully agree this naming convention.

Reply:
That's one of those "d'oh" moments. @tbursztyka created half a dozen of of "cleanup" tickets requesting following such a naming convention in various modules, and we've been working on them for a few months already. And now turns out we don't even have agreement on that naming convention.

Anyway, trying to find a way forward: you could name it arp_gratuitous_arp(), arp_gratuitous_req(), arp_gratuitous_msg() but IMHO those aren't really improvements over just arp_gratuitous().

Reply:
> trying to find a way forward: you could name it arp_gratuitous_arp(), arp_gratuitous_req(), arp_gratuitous_msg() but IMHO those aren't really improvements over just arp_gratuitous().

No need for this madness, the function name can be set to `arp_gratuitous()` if needed, this is minor issue. I just do not see the point in this naming convention.